contracts/src/NativeTokenSource.sol

```diff
@@ -0,0 +1,80 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {TeleporterTokenSource} from "./TeleporterTokenSource.sol";
+import {INativeTokenBridge} from "./interfaces/INativeTokenBridge.sol";
+import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/utils/SafeERC20.sol";
+import {SendTokensInput} from "./interfaces/ITeleporterTokenBridge.sol";
+import {IWrappedNativeToken} from "./interfaces/IWrappedNativeToken.sol";
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @title NativeTokenSource
+ * @notice This contract is an {INativeTokenBridge} that sends native tokens to another chain's
+ * {ITeleporterTokenBridge} instance, and gets represented by the tokens of that destination
+ * token bridge instance.
+ */
+contract NativeTokenSource is INativeTokenBridge, TeleporterTokenSource {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice The wrapped native token contract that represents the native tokens on this chain.
+     */
+    IWrappedNativeToken public immutable token;
+
+    /**
+     * @notice Initializes this source token bridge instance to send
+     * tokens to the specified destination chain and token bridge instance.
```

Comment:
This comment seems outdated since there is no destination specified in the constructor.